addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -57,20 +55,36 @@ def action_assign(self):
         Recompute batch state. In theory this is not necessary
         but the constraint on state has not proven to work correctly.
         """
-        super(StockPicking, self).action_assign()
+        super().action_assign()
         self.batch_id._compute_state()
 
+    @api.model
+    def create(self, vals):
+        """
+        If we create a batch and assign it to a picking while we are creating a picking then the picking type of
+        the batch will not be set by Odoo14 stock_picking_batch module during picking creation because in the UI
+        you cannot create a batch while you create a picking, but you can only assign a picking later to a batch.
+        Solve the issue by setting the batch picking type here.
+        """
+        res = super().create(vals)
+        if vals.get("batch_id"):
+            if not res.batch_id.picking_type_id:
+                res.batch_id.picking_type_id = res.picking_type_id
+                # Call sanity check again because we changed the allowed picking types
+                # by assigning a picking type to the batch
+                res.batch_id._sanity_check()
+        return res
+
     def write(self, vals):
         """
         If writing picking, check if previous batch is now complete.
         This will be used to trigger recompute of the batch state
         we can't relate on the state after as batch_id might be
         removed during write.
         """
-
         if self.batch_id:
             self = self.with_context(orig_batches=self.batch_id)
-        return super(StockPicking, self).write(vals)
+        return super().write(vals)
```

Comment:
Mmmm, I would prefer having more people weight on this. I don't feel confident going either way. I looked at Odoo's code and it uses super() so I thought it would be good to follow Odoo.

Reply:
Personally I would use `super()` - the old way should only be used if there is something unusual going on.